Source/Plugins/Core/com.equella.core/js/tsrc/settings/SettingsPage.tsx

```diff
@@ -0,0 +1,203 @@
+import { makeStyles } from "@material-ui/styles";
+import AdminDownloadDialog from "../settings/AdminDownloadDialog";
+import { ReactElement } from "react";
+import UISettingEditor from "./UISettingEditor";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import { generateFromError } from "../api/errors";
+import { AxiosError } from "axios";
+
+const useStyles = makeStyles((theme: Theme) => {
+  return {
+    heading: {
+      fontSize: theme.typography.pxToRem(15),
+      flexBasis: "33.33%",
+      flexShrink: 0
+    },
+    secondaryHeading: {
+      fontSize: theme.typography.pxToRem(15)
+    },
+    progress: {
+      display: "flex",
+      marginTop: theme.spacing.unit * 4,
+      justifyContent: "center"
+    }
+  };
+});
+
+interface SettingsPageProps extends TemplateUpdateProps {
+  refreshUser: () => void;
+}
+
+interface SettingCategory {
+  id: string;
+  details: {
+    name: string;
+    desc: string;
+  };
+}
+
+interface SettingGroup {
+  category: SettingCategory;
+  settings: GeneralSetting[];
+}
+
+const SettingsPage = (props: SettingsPageProps) => {
+  const classes = useStyles();
+  const { refreshUser, updateTemplate } = props;
+  const SETTING_CATEGORIES: SettingCategory[] = [
+    { id: "general", details: languageStrings.settings.general },
+    { id: "integration", details: languageStrings.settings.integration },
+    { id: "diagnostics", details: languageStrings.settings.diagnostics },
+    { id: "searching", details: languageStrings.settings.searching },
+    { id: "ui", details: languageStrings.settings.ui }
+  ];
+
+  const [settings, setSettings] = React.useState<GeneralSetting[]>([]);
+  const [adminDialogOpen, setAdminDialogOpen] = React.useState<boolean>(false);
+  const [loading, setLoading] = React.useState<boolean>(true);
+
+  // Update the title of App Bar
+  React.useEffect(() => {
+    updateTemplate(templateDefaults(languageStrings["com.equella.core"].title));
+  }, []);
+
+  // Fetch settings from the server
+  React.useEffect(() => {
+    let cancel = false;
+    fetchSettings()
+      .then(result => {
+        if (!cancel) {
+          setSettings(result.data);
+        }
+      })
+      .catch(error => {
+        handleError(error);
+      })
+      .finally(() => setLoading(false));
+
+    return () => {
+      cancel = true;
+    };
+  }, []);
+
+  const handleError = (error: AxiosError) => {
+    updateTemplate(templateError(generateFromError(error)));
+  };
+
+  // Group settings by their category and sort each group by setting name
+  const settingGroups = (): SettingGroup[] => {
+    const groups: SettingGroup[] = [];
+
+    SETTING_CATEGORIES.forEach(category => {
+      const settingsOfCategory = settings
+        .filter(setting => setting.group === category.id)
+        .sort((s1, s2) => {
+          return s1.name > s2.name ? 1 : -1;
+        });
+      groups.push({ category: category, settings: settingsOfCategory });
+    });
```

Comment:
could `forEach` be replaced with a `map`?

```suggestion
    return SETTING_CATEGORIES.map(category => {
      const settingsOfCategory = settings
        .filter(setting => setting.group === category.id)
        .sort((s1, s2) => {
          return s1.name > s2.name ? 1 : -1;
        });
      return { category: category, settings: settingsOfCategory };
    });
```